lib/tozny/realm.rb

```diff
@@ -216,6 +217,32 @@ def otp_challenge(type = nil, destination = nil, presence = nil, data = nil)
       else
         request_obj[:presence] = presence
       end
+
+      request_obj[:context] = context unless context.nil?
+
+      raw_call request_obj
+    end
+
+    # Create a magic link challenge session
+    #
+    # @param [String] destination The email address or phone number to which we will send a challenge
+    # @param [String] context     One of "verify," "authenticate," or "enroll"
+    # @param [String] callback    URL to which Tozny should submit signed link verification
+    # @param [String] hostname    Hostname for the generated OTP URL
+    # @param [Bool]   send        Flag whether or not to send the email (if false will return the OTP URL instead)
+    #
+    # @return [Hash] a hash of the session and presence for the challenge
+    #
+    def link_challenge(destination, context = nil, callback = nil, hostname = nil, send = true)
+      request_obj = {
+          method: 'realm.link_challenge',
+          destination: destination,
+          send: !! send ? 'yes' : 'no' # Hacky double-bang to convert nil to false and anything else into a boolean
```

Comment:
I always feel dirty when I use that. It qualifies as an explicit callout as "clever code" in many of the style guides I read and is usually frowned upon. But it made things so much easier here :-)